### PR TITLE
health check for status reason on searchrequest

### PR DESCRIPTION
**Pull request by rantifamilusi**
# Description

Ticket FAMS3-591, health check for status reason as part of the health diagnostic

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Tested locally

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: FAMS3-591

### CODE REVIEW

File: app/DynamicsAdapter/DynamicsAdapter.Web/Extensions/DynamicsAdapterExtension.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using DynamicsAdapter.Web.SearchRequest.Models;
+
+namespace DynamicsAdapter.Web.Extensions
+{
+    public static class DynamicsAdapterExtension
+    {
+        public static SearchRequestStatusReason  GetStatusReasonItem(this int reasonName)
```
Comment (peggy-gov-bc-ca) [inline app/DynamicsAdapter/DynamicsAdapter.Web/Extensions/DynamicsAdapterExtension.cs]: reasonName does not sound like an int.

File: app/DynamicsAdapter/DynamicsAdapter.Web/Health/StatusReasonHealthCheck.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                {
+                    Value = (int) reason,
+                    Label = new Label()
+                    {
+                        UserLocalizedLabel = new UserLocalizedLabel()
+                        {
+                            Label = reason.GetName()
+                        }
+                    }
+                }))
```
Comment (peggy-gov-bc-ca) [inline app/DynamicsAdapter/DynamicsAdapter.Web/Health/StatusReasonHealthCheck.cs]: })) is this compile?

File: app/DynamicsAdapter/DynamicsAdapter.Web/Startup.cs [hunk 3]
```diff
@@ -195,6 +197,7 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                 // registration of health endpoints see https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks
                 endpoints.MapHealthChecks("/health", new HealthCheckOptions
                 {
+                    AllowCachingResponses = false,
                     Predicate = _ => true,
```
Comment (peggy-gov-bc-ca) [inline app/DynamicsAdapter/DynamicsAdapter.Web/Startup.cs]: _ typo?

File: app/DynamicsAdapter/DynamicsAdapter.Web/Extensions/DynamicsAdapterExtension.cs (hunk 1 above)
Comment (rantifamilusi) [reply app/DynamicsAdapter/DynamicsAdapter.Web/Extensions/DynamicsAdapterExtension.cs]: resolved

File: app/DynamicsAdapter/DynamicsAdapter.Web/Health/StatusReasonHealthCheck.cs (hunk 2 above)
Comment (rantifamilusi) [reply app/DynamicsAdapter/DynamicsAdapter.Web/Health/StatusReasonHealthCheck.cs]: yes it compiles, are you getting an error ?

File: app/DynamicsAdapter/DynamicsAdapter.Web/Startup.cs (hunk 3 above)
Comment (rantifamilusi) [reply app/DynamicsAdapter/DynamicsAdapter.Web/Startup.cs]: No, it's a property name on the object (it's part of the healtcheck lib)

File: app/DynamicsAdapter/DynamicsAdapter.Web/Health/StatusReasonHealthCheck.cs (hunk 2 above)
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web/Health/StatusReasonHealthCheck.cs]: read carefully again. yes, it is correct.

**Pull request merged**